src/main/java/com/openlattice/auth0/CookieOrBearerSecurityContextRepository.java

```diff
@@ -68,36 +72,25 @@ public boolean containsContext( HttpServletRequest request ) {
     }
 
     private String tokenFromRequest( HttpServletRequest request ) {
-        //        final String value = request.getHeader( "Authorization" );
-        //
-        //        if ( value == null || !value.toLowerCase().startsWith( "bearer" ) ) {
-        //            return null;
-        //        }
-        //
-        //        String[] parts = value.split( " " );
-        //
-        //        if ( parts.length < 2 ) {
-        //            return null;
-        //        }
-        //
-        //        return parts[ 1 ].trim();
-        //
-        //        final String authorizationHeader = httpRequest.getHeader( "authorization" );
-        //
-        //        if ( authorizationHeader == null && authorizationCookie == null ) {
-        //            return null;
-        //        }
 
         final String authorizationInfo;
         final String authorizationHeader = getAuthorizationTokenFromHeader( request );
 
         if ( authorizationHeader != null ) {
             authorizationInfo = authorizationHeader;
         } else {
-            authorizationInfo = getAuthorizationTokenFromCookie( request );
+            authorizationInfo = getRequestCookie( request, AUTHORIZATION_COOKIE );
+            final String csrfTokenFromCookie = getRequestCookie( request, CSRF_COOKIE );
+            final String csrfTokenFromParams = request.getParameter( CSRF_COOKIE );
+            if ( csrfTokenFromCookie == null || csrfTokenFromParams == null ) {
+                return null;
+            }
+            if ( !StringUtils.equals( csrfTokenFromCookie, csrfTokenFromParams ) ) {
+                return null;
+            }
         }
 
-        if ( authorizationInfo == null || !authorizationInfo.toLowerCase().startsWith( "bearer" ) ) {
+        if ( authorizationInfo == null || !authorizationInfo.startsWith( BEARER_PREFIX ) ) {
```

Comment:
I have a vague recollection that this has broken in the past and we added in the toLowerCase to fix it.

Reply:
Thanks for flagging. I'll test locally and see if I run to any issues.

Reply:
After testing it locally, there doesn't seem to be issues with `toLowerCase` being removed. Any specific flow I should try?